pkg/kind/dashboard/kind_gen.go

```diff
@@ -0,0 +1,91 @@
+// This will be autogenerated, same as pkg/coremodel/dashboard/dashboard_gen.go is today
+
+package dashboard
+
+import (
+	"embed"
+
+	"github.com/grafana/grafana/pkg/framework/kind"
+	"github.com/grafana/thema"
+	"github.com/grafana/thema/vmux"
+)
+
+//go:embed kind.cue
+var cueFS embed.FS
+
+// TODO standard generated docs
+type Kind struct {
+	lin    thema.ConvergentLineage[*Entity]
+	meta   kind.CoreStructuredMeta
+	jendec vmux.Endec
+	valmux vmux.ValueMux[*Entity]
+}
+
+// type guard
+var _ kind.Structured = &Kind{}
+
+// TODO standard generated docs
+func NewKind(rt *thema.Runtime, opts ...thema.BindOption) (*Kind, error) {
+	kdef, err := kind.ParseKindFS(cueFS, rt)
+	if err != nil {
+		return nil, err
+	}
+	lin, err := kdef.BindKindLineage(rt, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	k := &Kind{
+		meta: kdef.Meta.(kind.CoreStructuredMeta),
+	}
+
+	// Get the thema.Schema that the meta says is in the current version (which
+	// codegen ensures is always the latest)
+	cursch := thema.SchemaP(lin, k.meta.CurrentVersion)
+	tsch, err := thema.BindType[*Entity](cursch, &Entity{})
+	if err != nil {
+		// Should be unreachable, modulo bugs in the Thema->Go code generator
+		return nil, err
+	}
+
+	k.jendec = vmux.NewJSONEndec("dashboard.json")
+	k.lin = tsch.ConvergentLineage()
+	k.valmux = vmux.NewValueMux(k.lin.TypedSchema(), k.jendec)
+	return k, nil
+}
+
+// TODO standard generated docs
+func (k *Kind) Name() string {
+	return "dashboard"
+}
+
+// TODO standard generated docs
+func (k *Kind) Lineage() thema.Lineage {
+	return k.lin
+}
+
+// TODO standard generated docs
+func (k *Kind) ConvergentLineage() thema.ConvergentLineage[*Entity] {
+	return k.lin
+}
+
+// JSONValueMux is a version multiplexer that maps a []byte containing JSON data
+// at any schematized dashboard version to an instance of *Entity.
+//
+// Validation and translation errors emitted from this func will identify the
+// input bytes as "dashboard.json".
+//
+// This is a thin wrapper around Thema's [vmux.ValueMux].
+func (k *Kind) JSONValueMux(b []byte) (*Entity, thema.TranslationLacunas, error) {
+	return k.valmux(b)
+}
```

Comment:
This is really the :chef-kiss: of all the new thema generics changes - a [`ConvergentLineage`](https://pkg.go.dev/github.com/grafana/thema#ConvergentLineage) encapsulates what today exists as [`coremodel.Interface`](https://github.com/grafana/grafana/blob/5641029a4afe3c368ecf6d38ee36c998083acc07/pkg/framework/coremodel/interface.go#L11-L29) into something agnostic wrt Grafana, tighter, and actually helped out by Go's type system.

It's easy to imagine this generated `JSONValueMux` method being integrated into the new entity api, as well as into the existing web framework, generically replacing the one-off validator we've written [here](https://github.com/grafana/grafana/blob/5641029a4afe3c368ecf6d38ee36c998083acc07/pkg/api/dashboard.go#L369-L380).

That said, this is also an interesting case because, [as these lines indicate](https://github.com/grafana/grafana/blob/5641029a4afe3c368ecf6d38ee36c998083acc07/pkg/api/dashboard.go#L369-L373), the dashboard schema is only expected to apply _after_ a certain schema version. We may need to adapt the code generator to allow an additional function to be injected into this value mux which performs a similar check. (Or maybe we can handle it entirely within the schema itself)